lib/ProductOpener/KnowledgePanels.pm

```diff
@@ -1065,6 +1068,54 @@ sub create_additives_panel($$$) {
 }
 
 
+=head2 create_ingredients_analysis_panel ( $product_ref, $target_lc, $target_cc )
+
+Creates a knowledge panel with the results of ingredients analysis.
+
+=head3 Arguments
+
+=head4 product reference $product_ref
+
+Loaded from the MongoDB database, Storable files, or the OFF API.
+
+=head4 language code $target_lc
+
+Returned attributes contain both data and strings intended to be displayed to users.
+This parameter sets the desired language for the user facing strings.
+
+=head4 country code $target_cc
+
+=cut
+
+sub create_ingredients_analysis_panel($$$) {
+
+	my $product_ref = shift;
+	my $target_lc = shift;
+	my $target_cc = shift;
+
+	$log->debug("create ingredients analysis panel", { code => $product_ref->{code} }) if $log->is_debug();
+
+    my $ingredients_analysis_data_ref = data_to_display_ingredients_analysis($product_ref);
+
+    if (defined $ingredients_analysis_data_ref) {
+
+        my $ingredients_analysis_panel_data_ref = {
+        };
```

Comment:
you should find another (shorter ?) variable name, because I didn't see the difference between `ingredients_analysis_data_ref` and `ingredients_analysis_panel_data_ref`.

Isn't `panel_data_ref` sufficient, given the limited scope ?

Reply:
Also you do not reference it but in last line ? Seems strange in this case that you need this.

Reply:
We don't need it indeed.